### ISSUE TITLE
Questions about the metrics computing

### ISSUE THREAD

**Issue by LeiTong02** [gulnazaki/counterfactual-benchmark#1]
Dear Authors,

Thank you for your excellent work on benchmarking various causal methods. I truly appreciate the clarity and depth of your paper and codebase. However, I have a couple of questions and suggestions regarding the implementation:

1. On the composition metric (L1 image space):
I noticed that the L1 loss is computed between input and reconstructed images in uint8 format. This may introduce unexpected behavior due to unsigned integer wrapping. For instance:

torch.tensor(230, dtype=torch.uint8) - torch.tensor(255, dtype=torch.uint8)
Output: tensor(231, dtype=torch.uint8)
This overflow leads to incorrect L1 distances. I encountered this issue in my own experiments on the CelebA dataset:

Using uint8, the L1 loss was above 150.

Using int (either torch.int or np.int), the value dropped to below 20, which seems more reasonable.

Would you consider converting images to a signed integer or floating-point type before computing the L1 distance to ensure correctness?

2. On the CLD metric:
In the paper, you mention using a pretrained unconditional VAE to compute the CLD metric. However, I could not find the corresponding checkpoint in the repository. Would you mind uploading the model weights or providing a link to download them?

Thanks again for the great work and looking forward to your response!

**Comment by gulnazaki** [comment]
Nice catch! Thanks for pointing this out, it has been patched in [this commit](https://github.com/gulnazaki/counterfactual-benchmark/commit/8c57c49c61475bb0c10ed3e713f7477079d70b52).

You can download the checkpoints for all datasets here: https://drive.google.com/drive/folders/1SAak8-S3HOCmPyk8JFojAjV-xv2spSlY?usp=sharing

**Comment by LeiTong02** [comment]
Hi @gulnazaki ,
 I encountered an additional issue with the ADNI dataset. I followed your preprocessing steps and used the same dataset class to load the data. However, I obtained 26,420 training images (previously it was 26,800 before applying indices_to_remove) and a test set of over 5,000 images — which differs from the numbers reported in your paper (10,780 training images from 180 subjects, and a test set of 2,240 images).

May I ask if there were any additional filtering steps applied to further reduce the number of images? I also used the middle 20 MRI slices and did not modify the code.

Best Regards,

**Comment by gulnazaki** [comment]
Hey @LeiTong02, so we run the final experiment with 10 slices after all, but we did not change this in the Appendix and the config (I am changing it in the code now). This cuts the number of images in half. As far as I know, we have not used any other preprocessing steps than the ones described in https://github.com/gulnazaki/counterfactual-benchmark/blob/main/counterfactual_benchmark/datasets/adni/preprocessing/README.md
